cmd/encryption-v1.go

```diff
@@ -80,6 +80,27 @@ func hasServerSideEncryptionHeader(header http.Header) bool {
 	return crypto.S3.IsRequested(header) || crypto.SSEC.IsRequested(header)
 }
 
+func isMultipart(objInfo ObjectInfo) bool {
+	if len(objInfo.Parts) == 0 {
+		return false
+	}
+	if !crypto.IsMultiPart(objInfo.UserDefined) {
+		return false
+	}
+	for _, part := range objInfo.Parts {
+		_, err := sio.DecryptedSize(uint64(part.Size))
+		if err != nil {
+			return false
+		}
+	}
+	// This is a further check to ensure that this object
+	// is uploaded using multipart mechanism.
+	if len(objInfo.ETag) == 32 {
```

Comment:
`` return len(objInfo.ETag) != 32 ``

Reply:
Also, this might not work for gateway when it sends ETag with `-1` for all operations:-). It is better to look for "-1" 

```
('-' not in md5_from_s3)
```

Reply:
> Also, this might not work for gateway when it sends ETag with `-1` for all operations:-). It is better to look for "-1"
> 
> ```
> ('-' not in md5_from_s3)
> ```

@harshavardhana I added backendType in ObjectInfo{} structure, that field which will be filled in xlMeta.ToObjectInfo()

This way we can target this specific use case